perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -1036,6 +1036,39 @@ public RichMember createSponsoredMember(PerunSession session, Vo vo, String name
 		return membersManagerBl.getRichMember(session, membersManagerBl.createSponsoredMember(session, vo, namespace, guestName, password, sponsor, true));
 	}
 
+	@Override
+	public RichMember setSponsorshipForMember(PerunSession session, Member sponsoredMember, User sponsor) throws InternalErrorException, MemberNotExistsException, AlreadySponsoredMemberException, UserNotInRoleException, PrivilegeException {
+		Utils.checkPerunSession(session);
+		getPerunBl().getMembersManagerBl().checkMemberExists(session, sponsoredMember);
+
+		if (sponsor == null) {
+			//sponsor is the caller
+			sponsor = session.getPerunPrincipal().getUser();
+		}
+
+		//only Perun Admin should has rights to do this operation
+		if (!AuthzResolver.isAuthorized(session, Role.PERUNADMIN)) {
+			throw new PrivilegeException(session, "Only PerunAdmin should have rights to call this method.");
+		}
+
+		//set member to be sponsored
+		return membersManagerBl.getRichMember(session, membersManagerBl.setSponsorshipForMember(session, sponsoredMember, sponsor));
+	}
+
+	@Override
+	public RichMember unsetSponsorshipForMember(PerunSession session, Member sponsoredMember) throws MemberNotExistsException, MemberNotSponsoredException, InternalErrorException, PrivilegeException {
+		Utils.checkPerunSession(session);
+		getPerunBl().getMembersManagerBl().checkMemberExists(session, sponsoredMember);
+
+		//only Perun Admin should has rights to do this operation
+		if (!AuthzResolver.isAuthorized(session, Role.PERUNADMIN)) {
+			throw new PrivilegeException(session, "Only PerunAdmin should have rights to call this method.");
+		}
+
+		//set member to be sponsored
```

Comment:
comment should be "unset sponsorship for member"

Reply:
fixed